src/MAUI/Maui.Samples/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs

```diff
@@ -192,16 +192,28 @@ private void ToolPicker_SelectedIndexChanged(object sender, EventArgs e)
             PointButton.IsEnabled = MultipointButton.IsEnabled = !_geometryEditor.IsStarted && tool is VertexTool;
         }
 
-        // Set the scale mode of the geometry editor.
+        // Set the scale mode for every geometry editor tool.
         private void CheckBox_CheckedChanged(object sender, CheckedChangedEventArgs e)
         {
-            // Determine if shape tools should use uniform scaling.
-            GeometryEditorScaleMode shouldBeUniform = (sender as CheckBox).IsChecked ? GeometryEditorScaleMode.Uniform : GeometryEditorScaleMode.Stretch;
+            // Determine the newly selected scale mode.
+            GeometryEditorScaleMode scaleMode =
+                (sender as CheckBox).IsChecked ? GeometryEditorScaleMode.Uniform : GeometryEditorScaleMode.Stretch;
 
-            // Update all shape tools scaling options.
-            foreach (ShapeTool tool in _toolDictionary.Values.Where(v => v is ShapeTool))
+            // Update the scale mode for every tool.
+            foreach (GeometryEditorTool tool in _toolDictionary.Values)
             {
-                tool.Configuration.ScaleMode = shouldBeUniform;
+                if (tool is FreehandTool freehandTool)
+                {
+                    freehandTool.Configuration.ScaleMode = scaleMode;
+                }
+                else if (tool is VertexTool vertexTool)
+                {
+                    vertexTool.Configuration.ScaleMode = scaleMode;
+                }
+                else
```

Comment:
may be keep this as else if in case a new type is added and goes to this fallback `tool as ShapeTool` will be null.